File: scripts/bulk_cleanup.js
const fs = require("fs");
const path = require("path");

// Setup paths
const SKILLS_DIR = path.join(__dirname, "../skills");
const DISABLED_DIR = path.join(__dirname, "../skills/.disabled");

// Ensure disabled directory exists
if (!fs.existsSync(DISABLED_DIR)) {
  try {
    fs.mkdirSync(DISABLED_DIR, { recursive: true });
  } catch (e) {
    console.error("Error creating disabled directory:", e);
    process.exit(1);
  }
}

// 1. ALWAYS KEEP these priorities (Keyword in name saves it)
const KEEP_KEYWORDS = [
  // Core User Stack
  "javascript",
  "js-",
  "-js",
  "typescript",
  "ts-",
  "node",
  "react",
  "next",
  "python",
  // Agents & AI
  "agent",
  "ai-",
  "llm",
  "rag-",
  "prompt",
  "robot",
  "bot-",
  "chat",
  "voice",
  "vision",
  "cortex",
  "brain",
  "memory",
  // Security (Explicitly requested)
  "security",
  "pentest",
  "hack",
  "attack",
  "vuln",
  "audit",
  "red-team",
  "defense",
  "shodan",
  "burp",
  "owasp",
  "xss",
  "injection",
  // Architecture & DB
  "architect",
  "design",
  "pattern",
  "clean",
  "refactor",
  "database",
  "db-",
  "sql",
  "postgres",
  "supabase",
  "mongo",
  "data",
  // General Productivity
  "git",
  "plan",
  "doc",
  "spec",
  "debug",
  "test",
  "review",
  "workflow",
  "manage",
  "product",
  "startup",
  "brainstorm",
  // Tools
  "docker",
  "k8s",
  "kubernetes",
  "terraform",
  "linux",
  "bash",
  "shell",
  "powershell",
  "windows",
  "terminal",
];

// 2. EXPLICITLY REMOVE these (unless saved above)
const REMOVE_KEYWORDS = [
  // Languages
  "ruby",
  "php",
  "java-",
  "javax",
  "c-pro",
  "cpp-",
  "cplusplus",
  "csharp",
  "dotnet",
  "fsharp",
  "scala",
  "haskell",
  "elixir",
  "julia",
  "swift",
  "kotlin",
  "dart",
  "ocaml",
  "r-pro",
  "matlab",
  "rust",
  "golang",
  "go-",
  "perl",
  "lua",
  "assembly",
  "fortran",
  // Mobile
  "android",
  "ios-",
  "flutter",
  "xamarin",
  "ionic",
  // Game Dev
  "unity",
  "unreal",
  "godot",
  "game",
  "minecraft",
  "glsl",
  "shader",
  // Platforms/Enterprise
  "salesforce",
  "sap",
  "oracle",
  "shopify",
  "wordpress",
  "moodle",
  "drupal",
  "magento",
  "joomla",
  "angular",
  "vue",
  "svelte",
  "ember",
  "backbone",
  "jquery",
  // Crypto
  "blockchain",
  "web3",
  "nft",
  "solidity",
  "crypto",
  "defi",
  "smart-contract",
  "ethereum",
];

function isKept(name) {
  name = name.toLowerCase();
  return KEEP_KEYWORDS.some((kw) => name.includes(kw));
}

function isRemoved(name) {
  name = name.toLowerCase();
  return REMOVE_KEYWORDS.some((kw) => name.includes(kw));
}

function main() {
  console.log(`Scanning ${SKILLS_DIR}...`);
  let movedCount = 0;
  let keptCount = 0;

  if (!fs.existsSync(SKILLS_DIR)) {
    console.error(`Error: ${SKILLS_DIR} does not exist`);
    return;
  }

  const items = fs.readdirSync(SKILLS_DIR);

  for (const item of items) {
    const itemPath = path.join(SKILLS_DIR, item);

    // Skip files and hidden directories
    if (item.startsWith(".")) continue;
    try {
      if (!fs.statSync(itemPath).isDirectory()) continue;
    } catch (e) {
      console.warn(`Could not stat ${item}: ${e.message}`);
      continue;
    }

    if (isKept(item)) {
      keptCount++;
      continue;
    }

    if (isRemoved(item)) {
      console.log(`Disabling: ${item}`);
      const targetPath = path.join(DISABLED_DIR, item);

      try {
        if (fs.existsSync(targetPath)) {
          console.log(
            `  Warning: Target ${targetPath} already exists. Skipping.`,
          );
        } else {
          fs.renameSync(itemPath, targetPath);
          movedCount++;
        }
      } catch (e) {
        console.error(`  Error moving ${item}: ${e.message}`);
      }
    } else {
      // Did not match Keep OR Remove -> Keep by default
      keptCount++;
    }
  }

  console.log("-".repeat(30));
  console.log("Summary:");
  console.log(`Disabled: ${movedCount} skills`);
  console.log(`Kept:     ${keptCount} skills`);
  console.log("-".repeat(30));
}

main();

File: scripts/bulk_cleanup.py
import os
import shutil
from pathlib import Path

# Setup paths
SKILLS_DIR = Path(r"c:\dev\antigravity-awasome-skills\.agent\skills\skills")
DISABLED_DIR = SKILLS_DIR.parents[0] / "skills/.disabled"

# Ensure disabled directory exists
if not DISABLED_DIR.exists():
    DISABLED_DIR.mkdir(parents=True)

# 1. ALWAYS KEEP these priorities (Keyword in name saves it)
# User: "Agentes, Next.js, Arquitectura, Seguridad, JavaScript/TypeScript, Python"
KEEP_KEYWORDS = [
    # Core User Stack
    "javascript", "js-", "-js", "typescript", "ts-", "node", "react", "next", "python",
    # Agents & AI
    "agent", "ai-", "llm", "rag-", "prompt", "robot", "bot-", "chat", "voice", "vision", "cortex", "brain", "memory",
    # Security (Explicitly requested)
    "security", "pentest", "hack", "attack", "vuln", "audit", "red-team", "defense", "shodan", "burp", "owasp", "xss", "injection",
    # Architecture & DB
    "architect", "design", "pattern", "clean", "refactor", "database", "db-", "sql", "postgres", "supabase", "mongo", "data",
    # General Productivity
    "git", "plan", "doc", "spec", "debug", "test", "review", "workflow", "manage", "product", "startup", "brainstorm",
    # Tools User might use
    "docker", "k8s", "kubernetes", "terraform", "linux", "bash", "shell", "powershell", "windows", "terminal"
]

# 2. EXPLICITLY REMOVE these (unless saved above)
# User: "Ruby, PHP, Java, C#, Unity" + others to reduce noise
REMOVE_KEYWORDS = [
    # Languages
    "ruby", "php", "java-", "javax", "c-pro", "cpp-", "cplusplus", "csharp", "dotnet", "fsharp",
    "scala", "haskell", "elixir", "julia", "swift", "kotlin", "dart", "ocaml", "r-pro", "matlab",
    "rust", "golang", "go-", # Assuming not needed unless specifically requested (Python/JS is main)
    "perl", "lua", "assembly", "fortran",
    # Mobile (Native)
    "android", "ios-", "flutter", "xamarin", "ionic", 
    # Game Dev
    "unity", "unreal", "godot", "game", "minecraft", "glsl", "shader",
    # Platforms/CMS/Enterprise
    "salesforce", "sap", "oracle", "shopify", "wordpress", "moodle", "drupal", "magento", "joomla",
    "angular", "vue", "svelte", "ember", "backbone", "jquery",
    # Crypto
    "blockchain", "web3", "nft", "solidity", "crypto", "defi", "smart-contract", "ethereum",
    # Specific Cloud (if not generic) - Keeping these is usually okay, but user has supabase. 
    # I'll be aggressive with obscure ones, but maybe keep AWS/Azure/GCP general skills just in case.
    # "aws", "azure", "gcp" -> I will NOT put them in REMOVE list, so they stay by default (Safety net)
]

def is_kept(name):
    for kw in KEEP_KEYWORDS:
        if kw in name:
            return True
    return False

def is_removed(name):
    for kw in REMOVE_KEYWORDS:
        if kw in name:
            return True
    return False

def main():
    print(f"Scanning {SKILLS_DIR}...")
    moved_count = 0
    kept_count = 0
    
    if not SKILLS_DIR.exists():
        print(f"Error: {SKILLS_DIR} does not exist")
        return

    for item in os.listdir(SKILLS_DIR):
        item_path = SKILLS_DIR / item
        if not item_path.is_dir():
            continue
        if item.startswith('.'):
            continue
            
        name = item.lower()
        
        # Logic: 
        # 1. If it matches KEEP keywords -> KEEP
        # 2. If it matches REMOVE keywords -> REMOVE
        # 3. Else -> KEEP (Default safe)
        
        if is_kept(name):
            kept_count += 1
            # print(f"Keeping: {item}")
            continue
            
        if is_removed(name):
            print(f"Disabling: {item}")
            target = DISABLED_DIR / item
            
            # Move it
            try:
                if target.exists():
                    print(f"  Warning: Target {target} already exists. Skipping.")
                else:
                    shutil.move(str(item_path), str(target))
                    moved_count += 1
            except Exception as e:
                print(f"  Error moving {item}: {e}")
        else:
            kept_count += 1
            # print(f"Keeping (Default): {item}")

    print("-" * 30)
    print(f"Summary:")
    print(f"Disabled: {moved_count} skills")
    print(f"Kept:     {kept_count} skills")
    print("-" * 30)
    print("To undo, use: python scripts/restore_all.py (you'd need to create it) or use skills_manager.py enable <name>")

if __name__ == "__main__":
    main()
